### DISCUSSION TITLE
Please enhance implicit operator support

### DISCUSSION THREAD

**Discussion by taori** [dotnet/csharplang#2687] [Language Ideas]
As pointed out here: https://github.com/aspnet/Mvc/issues/7724#issue-319272546 it would be great if the implicit operator would support downcasting to interfaces it's implementing.

The syntax i desire to write is visible here:

![grafik](https://user-images.githubusercontent.com/5545184/39485318-b27334dc-4d78-11e8-81f9-66607f7cb986.png)

@mkArtakMSFT tells me that this is a language issue. I was wondering what team of roslyn thinks about this.

**Comment by CyrusNajmabadi** [comment]
What are the types/declaration-code for "Ok" and "BadRequest"?  it's hard to do anything with this given hte lack of that info.

**Comment by taori** [comment]
@CyrusNajmabadi  

https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/OkObjectResult.cs

and 

https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/BadRequestResult.cs

Should have added that to begin with - you're right.

**Comment by HaloFour** [comment]
Related:  https://github.com/dotnet/csharplang/issues/563

**Comment by HaloFour** [comment]
https://github.com/dotnet/csharplang/issues/2460 ?
